### PR TITLE
plugins: add flag to 'download:plugins' script

### LINKED ISSUE #7122: plugins: add flag to 'download:plugins' script
**Description**

The main idea is to be able to call the `download:plugins` script with the ability to specify if the `.vsix` should be preserved or unpacked. 

The flag `preserveVsix` will:
- unpack the `.vsix` if `preserveVsix` is set to **false** (**default**)
- preserve plugins as `.vsix` if `preserveVsix` is set to **true**

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes #7122

The following pull-request adds a flag to the `download:plugins` script in order to control whether or not downloaded plugins should be **packed** or **unpacked**.

The **default** behavior is to **unpack** plugins (if no flag is provided), else an application may choose to specify whether or not to pack or unpack plugins using the `-packed (-p)` flag.

#### How to test

Call the `download:plugins` the following way:

| Command | Outcome |
|:---|:---|
| `yarn theia download:plugins` | should download plugins and unpack (not `.vsix`) |
| `yarn theia download:plugins -p=false` | should download plugins and unpack (not `.vsix`) |
| `yarn theia download:plugins -p=true` | should download plugins and <br/> **not** unpack (not `.vsix`) |

One can also specify the full name as such:

```bash
yarn theia download:plugins -packed=true
```

#### Review checklist

- [X] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: dev-packages/cli/src/theia.ts [hunk 1]
```diff
@@ -144,7 +144,16 @@ function rebuildCommand(command: string, target: ApplicationProps.Target): yargs
         })
         .command({
             command: 'download:plugins',
-            handler: () => downloadPlugins()
+            describe: 'Download defined external plugins.',
+            builder: {
+                'preserveVsix': {
```
Comment (akosyakov) [inline dev-packages/cli/src/theia.ts]: maybe better `unpack` which is false by default?

Comment (vince-fugnitto) [reply dev-packages/cli/src/theia.ts]: I think we discussed that perhaps we wanted the preservation of the `.vsix` file as to be the one which requires the flag to be turned on (requires an application to actually request such a feature), since most clients would it expect it to unpack?

The only client I know as of now that'd want to preserve is the `theia-electron` AppImage.
cc @marcdumais-work

Comment (akosyakov) [reply dev-packages/cli/src/theia.ts]: Ah, that’s fine I just wanted to have shorter option name ) but it is also not so important

Comment (vince-fugnitto) [reply dev-packages/cli/src/theia.ts]: I agree :) I updated the flag so it is shorter.
I use `-packed` or `-p` to control the behavior when downloading plugins.

Comment (marcdumais-work) [reply dev-packages/cli/src/theia.ts]: > The only client I know as of now that'd want to preserve is the `theia-electron` AppImage.

Not only AppImage but potentially all OS-specific packagings for Electron to be safe (.deb, .rpm, .msi, .exe, .dmg, ...). 

TL;DR: we noticed that the [Java Language Support for Visual Studio Code by Red Hat](https://github.com/redhat-developer/vscode-java) extension writes to the extension's directory at runtime. This fails if the extension is unpacked in such a way that the Electron app user does not have write access to it, which happens when installing a packaged Electron app (except maybe if packages as simple .tar or .zip and extracted by end-user). 

We are not sure if other VS Code extensions may also be also writing to their directory at runtime, so to be safe I would keep Electron-bundled extensions packed in `.vsix` format. 

(note: I did not see this behavior on writing in one's own folder at runtime for built-in extensions)

**Comment by vince-fugnitto** [comment]
Thank you @akosyakov @marcdumais-work !
